### PR TITLE
Add argument support for CLI and plugins

### PR DESCRIPTION
**Pull request by pmuens**
Add the possibility to access arguments inside plugins (e.g. "--function function1" or "-f function").
Additionally the "lifeCycleEvent" property is renamed to "lifecycleEvent".

### CODE REVIEW

File: lib/plugins/HelloWorld/HelloWorld.js [hunk 1]
```diff
@@ -20,20 +25,29 @@ class HelloWorld {
     };
   }
 
-  printGoodMorning() {
-    const message = 'Good morning';
+  printGoodMorning(args) {
+    let message = 'Good morning madam';
```
Comment (flomotlik) [inline lib/plugins/HelloWorld/HelloWorld.js]: Nice thing going with female by default, I like that!

Comment (pmuens) [inline lib/plugins/HelloWorld/HelloWorld.js]: That was actually intended!

**Comment by flomotlik** [comment]
GTM once github is back

File: lib/plugins/HelloWorld/HelloWorld.js (hunk 1 above)
Comment (flomotlik) [inline lib/plugins/HelloWorld/HelloWorld.js]: I know :D

**Pull request merged**